--- time-1.7.orig/configure.in  1996-06-12 10:07:06.000000000 -0700
+++ time-1.7/configure.in       2011-05-08 11:24:06.544268992 -0700
@@ -1,9 +1,6 @@
 dnl Process this file with autoconf to produce a configure script.
 AC_INIT(time.c)
-VERSION=1.7
-AC_SUBST(VERSION)
-PACKAGE=time
-AC_SUBST(PACKAGE)
+AM_INIT_AUTOMAKE(time, 1.7)

 AC_ARG_PROGRAM

@@ -15,7 +12,7 @@
 dnl Checks for header files.
 AC_HEADER_STDC
 AC_HEADER_SYS_WAIT
-AC_CHECK_HEADERS(unistd.h string.h sys/rusage.h)
+AC_CHECK_HEADERS(unistd.h string.h sys/rusage.h sys/resource.h)

 dnl Checks for typedefs, structures, and compiler characteristics.
 AC_C_CONST
